lib/mesa.js

```diff
@@ -562,9 +562,9 @@ Mesa.prototype = {
     });
   },
 
-  runQueue(queue, value) {
+  runQueue(queue, value, origin = value) {
     const context = this;
-    const reducer = (soFar, step) => soFar.then(step.bind(context));
+    const reducer = (soFar, step) => soFar.then(_.bind(step, context, _, origin));
```

Comment:
`Function.prototype.bind()` supports [adding arguments as well](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Function/bind). Do you really want to bind `_` - the library - to step?

Reply:
the underscore '_' does not bind the library, it is a placeholder.
https://lodash.com/docs/4.17.15#bind

looks like prototype.bind() doesn't support his.

Reply:
Oh, my JS mojo is fading. In this case a closure is probably better than this `bind()` construct.